ui/mirage/services/project.ts

```diff
@@ -23,3 +23,21 @@ export function get(schema: any, { requestBody }: Request): Response {
 
   return this.serialize(resp, 'application');
 }
+
+export function update(schema: any, { requestBody }: Request): Response {
+  let requestMsg = decode(UpsertProjectRequest, requestBody);
+  let name = requestMsg.getProject().getName();
+  let variablesList = requestMsg
+    .getProject()
+    .getVariablesList()
+    .map((v) => v.toObject());
+  let model = schema.projects.findBy({ name });
+
+  variablesList.forEach((v) => model.createVariable(v));
```

Comment:
I realized I was mistaken in my previous suggestion here. I forgot this was an upsert operation — sorry about that!

Maybe we can try something like this:

```suggestion
  model.variables = variablesList.map((v) => model.newVariable(v));
  model.save();
```